Add integration for Neuphonic TTS

adnansiddiquei opened the pull request:
This PR implements a service to use Neuphonic as a TTS provider. I am an engineer at Neuphonic.
- https://neuphonic.com/
- https://docs.neuphonic.com/quickstart

adnansiddiquei commented:
Awesome, I'll add in this example now!

adnansiddiquei commented:
I have added two examples and switched this PR into ready for review!

aconchillo commented on src/pipecat/services/neuphonic.py:
```diff
@@ -0,0 +1,365 @@
+    return result
+
+
+class NeuphonicTTSService(WordTTSService, WebsocketService):
```
Please use `WebsocketWordTTSService` instead.

aconchillo replied:
Actually, it seemsNeuphonic doesn't support word timestamps yet. In which case, we probably need `InterruptibleTTSService`. This will make sure interruptions work correctly.

aconchillo commented on src/pipecat/services/neuphonic.py:
```diff
@@ -0,0 +1,365 @@
+            sample_rate=sample_rate,
+            **kwargs,
+        )
+        WebsocketService.__init__(self)
```
remove.

aconchillo commented on src/pipecat/services/neuphonic.py:
```diff
@@ -0,0 +1,365 @@
+        params: InputParams = InputParams(),
+        **kwargs,
+    ):
+        WordTTSService.__init__(
```
Change to `super().__init__(...` once we change the base class.

adnansiddiquei commented:
Thanks @aconchillo I have corrected all of your raised review comments.

I have also removed the notion of "models" in the Neuphonic implementation as Neuphonic now routes requests to the respective model based on the provided `voice_id` to make things simpler. So the end user does not need to worry about which model they need to select. Therefore I have removed all `self.set_model_name(model)` calls, I hope this doesn't mess with the metrics or anything on the `pipecat` side?

aconchillo commented:
> Thanks @aconchillo I have corrected all of your raised review comments.
> 
> I have also removed the notion of "models" in the Neuphonic implementation as Neuphonic now routes requests to the respective model based on the provided `voice_id` to make things simpler. So the end user does not need to worry about which model they need to select. Therefore I have removed all `self.set_model_name(model)` calls, I hope this doesn't mess with the metrics or anything on the `pipecat` side?

Thank you! I think it should be OK.

aconchillo commented:
LGTM!
